dashboards/clicks_and_impressions.py

```diff
@@ -0,0 +1,64 @@
+import pandas as pd
+import streamlit as st
+from auxiliary_functions import extract_query_params, process_query_params
+
+st.title("Clicks & Impressions Dashboard")
+
+# Initialize connection
+conn = st.connection("mysql", type="sql")
+
+df = conn.query("select * from website_event;")
+
+df_processed = df.copy(deep=True)
+#
+df_processed["url"] = "/?" + df["url_query"].astype(str)
+df_processed["query_params"] = df_processed["url"].apply(extract_query_params)
+df_processed["parsed_params"] = df_processed["query_params"].apply(process_query_params)
```

Comment:
Since https://github.com/AdvisorySG/mentorship-streamlit/commit/9ef75bd5763d94f7d964c1bb48824a295782f00c has been merged, please use DuckDB SQL queries instead to retrieve the initial dataframe. Some sample code is available at https://github.com/AdvisorySG/mentorship-streamlit/issues/48.